sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -954,12 +954,12 @@ private class SortMergeFullOuterJoinScanner(
     }
 
     if (leftMatches.size <= leftMatched.capacity) {
-      leftMatched.clear()
+      leftMatched.clearUntil(leftMatches.size)
```

Comment:
I suspect this is a correct change as you describe it, but can you help me be 100% sure by describing why we know the bits above this point will be 0 or don't matter? I'm trying to think of a case where a leftover set 1 bit from previous computation causes a problem. Is that definitely not possible?